orm/src/main/java/org/hibernate/search/batchindexing/impl/IdentifierConsumerDocumentProducer.java

```diff
@@ -268,9 +268,7 @@ private void index(Object entity, Session session, InstanceInitializer sessionIn
 		DocumentBuilderIndexedEntity docBuilder = entityIndexBinding.getDocumentBuilder();
 		TwoWayFieldBridge idBridge = docBuilder.getIdBridge();
 		String idPropertyName = docBuilder.getIdPropertyName();
-		if ( idPropertyName != null ) {
-			conversionContext.pushProperty( docBuilder.getIdFieldName() );
-		}
+		conversionContext.pushProperty( docBuilder.getIdFieldName() );
```

Comment:
Looks like you roll this back in the following commit. Maybe you should remove this change from this commit?